### PR TITLE
support Update() method

### LINKED ISSUE #27: port to circleci 2.0
We're still using the old circle ci format

I will use the config.yml we have in hydra, which will also use govendor for managing the dependencies

### PR DESCRIPTION
**Pull request by batmany13**
closes #26 
closes #27 

minor clean up so that "Api" related code is in the api file

clean up on error response

refactored tests

### CODE REVIEW

**Comment by batmany13** [comment]
closing, as I am going include the circleci 2.0 as part of this

**Comment by martinatsynqfm** [comment]
@batmany13
> closes #9

#9 is a pull request

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+          name: Download Dependencies
+          command: |
+            sudo chown  -R circleci:circleci $SYNQ_DIR
+            go get github.com/kardianos/govendor
+            govendor sync
+      - run:
+          name: Run Golang Tests
+          command: |
+            GOPATH=$GOPATH:$SYNQ_DIR/$REPO/vendor overalls -project=github.com/SYNQfm/$REPO -covermode=count -debug -ignore=.git,vendor -- -race -v
+            goveralls -coverprofile=$SYNQ_DIR/$REPO/overalls.coverprofile -service circle-ci -repotoken $COVERALLS_TOKEN
```
Comment (martinatsynqfm) [inline .circleci/config.yml]: the shell variable references in this file are not `"${QUOTED}"`

**Comment by martinatsynqfm** [comment]
it seems to me that this pull request includes several changes that are not mentioned in #27:

- moving functions unrelated to `Update`
- altering circleci configuration
- adding `vendor/vendor.json` with no description of its purpose

**Comment by batmany13** [comment]
@martinatsynqfm yes #27 was not descriptive enough, its just a general move to 2.0, and I've created a standardized "circleci" config for Golang microservices, which uses govendor to manage the dependencies, so this is used in hydra as well.

The move is just minor clean up, where "api" related methods were not grouped together and is part of the Update() fix, since other related methods were not in the right file.

File: .circleci/config.yml (hunk 1 above)
Comment (batmany13) [inline .circleci/config.yml]: will change!

**Pull request merged**